Feature: Improve sequence and combo hotkey support

Wxh16144 opened the pull request:
- Fixed the issue where sequence hotkeys and combination hotkeys could not be used together.
- Fixed a bug where array-type sequence hotkeys (e.g. `["h>i", "o>k"])` would not work as expected.
- Refactored internal logic to reduce code redundancy and improve readability.
- Improved sequence matching logic to support overlapping and substring sequences.
- Enhanced test coverage for mixed and array sequence hotkey scenarios.

pksunkara commented:
@JohannesKlauss Please let me know if you need any help with reviewing and testing this.

pksunkara reviewed (approved):
Looks great! I have manually tested these changes too and everything is working.

Would love to get this PR merged since it fixes important bugs with sequential hotkeys.

pksunkara commented:
@Wxh16144 Can you please rebase this?

pksunkara commented on packages/react-hotkeys-hook/src/lib/useHotkeys.ts:
```diff
@@ -89,86 +118,66 @@ export default function useHotkeys<T extends HTMLElement>(
+          : mapCode(e.code)
 
-            recordedKeys = []
+        if (!isHotkeyModifier(currentKey.toLowerCase())) {
```
Need to keep the TODO comment about making modifiers work with sequences.

> // TODO: Make modifiers work with sequences

JohannesKlauss commented:
If you resolve the merge conflicts we can ship this :)

pksunkara commented:
@Wxh16144 Will you be able to find time to get this done? Or should I take over & create a new PR?

Wxh16144 commented:
Sorry, I've been too busy recently. I will take some time later to reorganize the PR content.

Wxh16144 committed:
Feature: Improve sequence and combo hotkey support

🧪 test: unit test

feat(hotkeys): refactor hotkey parsing to separate combo and sequence handling

chore: update unit test

feat(hotkeys): implement sequenceEndsWith function and refactor sequence handling in useHotkeys

feat(hotkeys): refactor hotkey parsing to handle combo and sequence hotkeys separately

chore: code style

update unit test

# Conflicts:
#	packages/react-hotkeys-hook/src/lib/useHotkeys.ts

Wxh16144 commented:
There were conflicts before, so I first cherry-picked the test cases and then asked AI `GPT-5.5` to help clean things up a bit. The resulting diff is a little larger than expected, but I did run the full test suite and everything is passing.

I’m not completely sure whether this approach is acceptable, so I’d really appreciate your review.

Thanks!
